### ISSUE TITLE
Support for degree Celsius and conversions from/to Kelvin

### ISSUE THREAD

**Issue by SimonHeybrock** [scipp/scipp#2889]
As far as I can tell degree Celsius are currently not supported by the string parsing? This should be fixed.

To make it actually useful, we have to support conversion from/to Kelvin. Currently we use `llnl::units::quick_convert`, which does not work in this case. We should switch to (or fall back to) `llnl::units::convert` in this case.

We should also check that we store the `e_flag` field when writing HDF5 files, or handle this in some other way.

**Comment by jl-wynen** [comment]
> We should also check that we store the e_flag field when writing HDF5 files, or handle this in some other way.

Why? We didn't add this in order to be less dependent on LLNL/Units.

**Comment by SimonHeybrock** [comment]
Yes, that is why I wrote "or handle this in some other way", i.e., instead of storing this specific flag we can consider storing it in another way. Just because we do not want to be tied too much to LLNL/units does not imply that we can decline adding support for degree Celsius?

**Comment by jl-wynen** [comment]
But why do we need the flag?

**Comment by SimonHeybrock** [comment]
The flag is how the library supports units such as degree Celsius.

**Comment by jl-wynen** [comment]
Ok. So we need to distinguish between errors and other reasons for setting `e_flag`.

**Comment by SimonHeybrock** [comment]
Note that parsing should be supported after #2931.
